bundles/org.openhab.core.addon/src/main/java/org/openhab/core/addon/AddonInfoRegistry.java

```diff
@@ -106,8 +106,8 @@ public void removeAddonInfoProvider(AddonInfoProvider addonInfoProvider) {
             builder.withName(b.getName());
             builder.withDescription(b.getDescription());
         }
-        if (!(a.isMasterAddonInfo() || b.isMasterAddonInfo())) {
-            builder.isMasterAddonInfo(false);
+        if (!a.isMasterAddonInfo() && b.isMasterAddonInfo()) {
+            builder.isMasterAddonInfo(true);
```

Comment:
Why is it important that `a` is not `masterAddonInfo`?

Reply:
Just to avoid the useless call to builder.isMasterAddonInfo when the builder value is already set to true, and have a similar pattern as the remaining in that method.

But could also be done differently by replacing the full if then block by just:
builder.isMasterAddonInfo(a.isMasterAddonInfo() || b.isMasterAddonInfo()};